src/debug-generator/debug-generator.c

```diff
@@ -216,14 +216,27 @@ static int process_unit_credentials(const char *credentials_dir) {
                         log_debug("Wrote unit file '%s' from credential '%s'", unit, de->d_name);
 
                 } else if (dropin) {
-                        r = write_drop_in(arg_dest, dropin, 50, "credential", d);
+                        _cleanup_free_ char *dropin_unit = NULL;
+                        const char *tilde, *dropin_name;
+
+                        tilde = strchrnul(dropin, '~');
+                        dropin_unit = strndup(dropin, tilde - dropin);
+                        dropin_name = tilde ? tilde + 1 : "50-credential";
```

Comment:
Either strchr should be used instead of strchrnul, or the check should be `!isempty(tilde)`?

Reply:
BTW, empty drop-in name should be refused, i.e. standalone `~`